src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertFrom-StringData.cs

```diff
@@ -55,6 +61,11 @@ protected override void ProcessRecord()
                 return;
             }
 
+            if (AsLiteral.IsPresent)
+            {
+                _stringData = Regex.Escape(_stringData);
```

Comment:
this approach causes an issue if the string looks like this:

```powershell
PS> 'a\b=c\d' | convertfrom-stringdata -aslit
Name                           Value
----                           -----
a\\b                           c\d
```

the regex escape should only be applied to the right side of the `=`

Reply:
@JamesWTruher Current behaviour with escaping yourself does the same thing:

```pwsh
('a\b=c\d' -replace '\\','\\') | ConvertFrom-StringData

Name                           Value
----                           -----
a\\b                           c\d
```

My understanding was the whole input string needs to be escaped, but I've noticed the value always being unescaped again in the code.

Reply:
Since `-AsLiteral` is expected to just escape the input, then the resulting behavior should be the same as if the user manually escaped the input themselves.  In this case, the behavior is the same even if a bit surprising.  I think it would be a different issue to discuss changing `Name` to not be escaped but would be a breaking change.